### ISSUE TITLE
Problems with installation

### ISSUE THREAD

**Issue by Olli781** [tiagohm/nebulosa#405]
Hello everyone,
when installing under UBUNTU 24.04, the following error message appears when running the script:

Build at: 2024-05-17T09:07:17.533Z - Hash: 6e9b3560450bd860 - Time: 99655ms

Warning: /home/astro1/Downloads/nebulosa-main/desktop/node_modules/@angular/core/fesm2022/core.mjs depends on 'rxjs'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/astro1/Downloads/nebulosa-main/desktop/node_modules/@angular/core/fesm2022/core.mjs depends on 'rxjs/operators'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/astro1/Downloads/nebulosa-main/desktop/src/app/atlas/atlas.component.ts depends on 'moment'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/astro1/Downloads/nebulosa-main/desktop/src/app/image/image.component.ts depends on 'interactjs'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/astro1/Downloads/nebulosa-main/desktop/src/app/image/image.component.ts depends on 'panzoom'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/astro1/Downloads/nebulosa-main/desktop/src/shared/components/map/map.component.ts depends on 'leaflet'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

  • electron-builder  version=24.13.3 os=6.8.0-1004-raspi
  • loaded configuration  file=/home/astro1/Downloads/nebulosa-main/desktop/electron-builder.json
  • writing effective config  file=release/builder-effective-config.yaml
  • packaging       platform=linux arch=arm64 electron=30.0.1 appOutDir=release/linux-arm64-unpacked
  • building        target=AppImage arch=arm64 file=release/nebulosa-0.1.0-arm64.AppImage
  • building        target=deb arch=arm64 file=release/nebulosa_0.1.0_arm64.deb
  ⨯ cannot execute  cause=exit status 1
                    errorOut=/home/astro1/.cache/electron-builder/fpm/fpm-1.9.3-2.3.1-linux-x86/lib/ruby/bin/ruby: Zeile 6: /home/astro1/.cache/electron-builder/fpm/fpm-1.9.3-2.3.1-linux-x86/lib/ruby/bin.real/ruby: Kann die Binärdatei nicht ausführen: Fehler im Format der Programmdatei
    /home/astro1/.cache/electron-builder/fpm/fpm-1.9.3-2.3.1-linux-x86/lib/ruby/bin/ruby: Zeile 6: /home/astro1/.cache/electron-builder/fpm/fpm-1.9.3-2.3.1-linux-x86/lib/ruby/bin.real/ruby: Erfolg
    
command=/home/astro1/.cache/electron-builder/fpm/fpm-1.9.3-2.3.1-linux-x86/fpm -s dir --force -t deb -d libgtk-3-0 -d libnotify4 -d libnss3 -d libxss1 -d libxtst6 -d xdg-utils -d libatspi2.0-0 -d libuuid1 -d libsecret-1-0 --deb-recommends libappindicator3-1 --deb-compression xz --architecture arm64 --after-install /tmp/t-As69GW/2-after-install --after-remove /tmp/t-As69GW/1-after-remove --description '
     The complete integrated solution for all of your astronomical imaging needs.' --version 0.1.0 --package /home/astro1/Downloads/nebulosa-main/desktop/release/nebulosa_0.1.0_arm64.deb --name nebulosa --maintainer 'Tiago Melo <tiago.henrique.cco@gmail.com>' --url https://tiagohm.github.io/nebulosa --vendor 'Tiago Melo <tiago.henrique.cco@gmail.com>' --deb-priority optional --license MIT /home/astro1/Downloads/nebulosa-main/desktop/release/linux-arm64-unpacked/=/opt/nebulosa /home/astro1/Downloads/nebulosa-main/desktop/dist/assets/icons/nebulosa-256.png=/usr/share/icons/hicolor/256x256/apps/nebulosa.png /tmp/t-As69GW/4-nebulosa.desktop=/usr/share/applications/nebulosa.desktop
                    workingDir=

Does anyone have an idea how I can fix this error? Thank you very much and best regards Oliver

**Comment by tiagohm** [comment]
Hi Oliver,

Could you check out this link https://www.beekeeperstudio.io/blog/electron-apps-for-arm-and-raspberry-pi e the related issue https://github.com/electron-userland/electron-builder/issues/3901#issuecomment-499121694 and see if it solves the error?

**Comment by Olli781** [comment]
Thank you very much for your help. 

Unfortunately, there are new error messages:
sudo apt install ruby ​​ruby-devsudo gem install fpm
Reading package lists… Done
Building dependency tree… Done
Status information is being read in… Done
E: Package ruby-devsudo cannot be found.
E: Package install cannot be found.
E: Package fpm cannot be found.

Is it perhaps easier to switch to a different Linux version?

**Comment by tiagohm** [comment]
Try running these commands:

```bash
sudo apt install ruby ruby-dev
sudo gem install fpm
```

Inside `desktop` folder, run this command to build .deb package:

```bash
USE_SYSTEM_FPM=true npm run electron:build:deb
```

I was able to build successfully on Raspberry Ubuntu 22.04

**Comment by tiagohm** [comment]
Hi Oliver,

I am pleased to inform you that the first version of the software has been released. You can now download and test it. I appreciate your patience.

I will now proceed to close the issue. If you have any additional questions, please feel free to reopen the issue.

**Comment by Olli781** [comment]
Hi Tiago,

 

First of all, thank you very much for your mail.

 

In the meantime, I installed AstroArch (https://github.com/devDucks/astroarch) on my Raspberry Pi 5. AstroArch is based on Archlinux. Unfortunately, as far as I know, Archlinux does not directly support installing .RPM or .DEB packages. Is there a easy way to make your installation packages readable for Archlinux and Pacman?

 

Thank you for your help and best regards,

 

Oliver

 

Von: Tiago ***@***.***> 
Gesendet: Freitag, 18. Oktober 2024 23:22
An: tiagohm/nebulosa ***@***.***>
Cc: Oliver ***@***.***>; Author ***@***.***>
Betreff: Re: [tiagohm/nebulosa] Problems with installation (Issue #405)

 

Hi Oliver,

I am pleased to inform you that the first version of the software has been released. You can now download and test it. I appreciate your patience.

I will now proceed to close the issue. If you have any additional questions, please feel free to reopen the issue.

**Comment by tiagohm** [comment]
Hi Oliver,

I just released the version [0.1.2](https://github.com/tiagohm/nebulosa/releases/tag/0.1.2). It is now possible to install on archlinux. Just download the `.pacman` artifact and install running `pacman -U nebulosa-0.1.2-linux-aarch64.pacman`
